Realm/RLMSyncUser.h

```diff
@@ -32,9 +34,34 @@ typedef NS_ENUM(NSUInteger, RLMSyncUserState) {
     RLMSyncUserStateError,
 };
 
+/**
+ Possible values for permissions that can be granted to a given user for working
+ with a given synced Realm.
+ */
+typedef NS_OPTIONS(NSUInteger, RLMSyncRealmPermission) {
+    /// The Realm can be read by a target user.
+    RLMSyncRealmPermissionRead,
+    /// The Realm can be modified by a target user.
+    RLMSyncRealmPermissionWrite,
+    /// The Realm can be managed by a target user.
+    RLMSyncRealmPermissionManage,
+};
+
 /// A block type used for APIs which asynchronously vend an `RLMSyncUser`.
 typedef void(^RLMUserCompletionBlock)(RLMSyncUser * _Nullable, NSError * _Nullable);
 
+/// A block type used to asynchronously report permission changes.
+typedef void(^RLMSyncPermissionChangeBlock)(RLMSyncManagementObjectStatus, NSError * _Nullable);
```

Comment:
I think it will be cumbersome to use these callbacks. If I am making a permission change or offer, I want to get a response for my action specific to that action. With this design all responses are grouped together and dispatched though these two callbacks. This means that if a user wants to handle a specific change or offer they need to do their own complicated bookkeeping.